### PR TITLE
Prevent group var overlap in differential abundance config

### PR DESCRIPTION
**Pull request by jernestmyers**
Resolves first item in https://github.com/VEuPathDB/web-monorepo/issues/358

These changes allow us to:
1. Pass in a list of values to disable in the checkbox list
2. Pass in content to be displayed in a tooltip re: disabled checkbox values

Note: I removed the `useMemo` from `ValuePicker` because I don't think it's necessary. I also had concerns about using the same `EMPTY_ARRAY` reference for each fallback, so I've made unique empty arrays for each prop instead.

### CODE REVIEW

**Review by dmfalke — APPROVED** [review]
This looks great to me. Just one thought about `Tootltip`!

File: packages/libs/coreui/src/components/inputs/checkboxes/CheckboxList.tsx [hunk 1]
```diff
@@ -187,16 +198,17 @@ export default function CheckboxList<T>({
...
-                  onChange={() => onChangeHandler(item.value)}
-                />{' '}
-                {item.display}
-              </label>
+              {item.disabled ? (
+                <Tooltip title={disabledCheckboxTooltipContent ?? ''}>
+                  <label style={{ color: 'lightgrey' }}>
+                    <input {...sharedInputAttributes} disabled /> {item.display}
+                  </label>
+                </Tooltip>
```
[the remark this answers is not in the document]
Comment (jernestmyers) [reply packages/libs/coreui/src/components/inputs/checkboxes/CheckboxList.tsx:206]: I had a similar concern, which is why I decided to add extra code to only render the tooltips for disabled items as opposed to wrapping all items in a `<Tooltip>` and passing in an empty string if the item is not disabled.

Do you suspect we'll have large enough checkbox lists and enough disabled values for this to cause any real concerns? @asizemore, you have any thoughts on this to share as well?

[the remark this answers is not in the document]
Comment (dmfalke) [reply packages/libs/coreui/src/components/inputs/checkboxes/CheckboxList.tsx:206]: Ah, that's a good point. I can't imagine we would run into such issues.

[the remark this answers is not in the document]
Comment (asizemore) [reply packages/libs/coreui/src/components/inputs/checkboxes/CheckboxList.tsx:206]: I dont know of any categorical vars with loads of values. The vars used most often are <10 for sure. The biggest one i've seen was some age var that is improperly categorical 😄 and it had 60 values.

[the remark this answers is not in the document]
Comment (jernestmyers) [reply packages/libs/coreui/src/components/inputs/checkboxes/CheckboxList.tsx:206]: Okay, I'm going to leave it as-is for now then. I'm not sure how often consumers of these components will use the optional `disabledValues` prop in the first place, nor if we'd ever have enough disabled values to impact performance!

**Pull request merged**